### PR TITLE
Add language guide for C#

### PR DESCRIPTION
**Pull request by glokta1**

### CODE REVIEW

File: docs/challenges/language-support/csharp.mdx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+To run the code for your challenge locally, you'll need `dotnet` installed. Our test runners use Dotnet 8.0 (as of April 2024).
+
+The script for your challenge (like `./your_sqlite3.sh`) will automatically compile your code using `dotnet` before executing it. It'll
+look something like this:
+
+```sh
+#!/bin/sh
+
+exec dotnet run --project . --configuration Release "$@"
```
Comment (rohitpaulk) [inline docs/challenges/language-support/csharp.mdx:16]:
```suggestion
exec dotnet run --project . --configuration Release -- "$@"
```

(We're changing this here: https://linear.app/codecrafters/issue/CC-1170/c-spawn-scripts-are-not-properly-passing-arguments-to-main)

File: docs/challenges/language-support/csharp.mdx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        Foo.foo();  // calls the method foo() from Foo class in Foo.cs
+    }
+}
+```
+
+</CodeGroup>
+
+By default, all classes in `src/` belong to the same namespace. This namespace is called the _project namespace_. E.g. If the project name is `MyProject`, all classes are defined under the `MyProject` namespace. This is why you can call `foo()` without importing anything.
+
+If you want to define a class under a _different_ namespace, you'll need to enclose it within braces like this:
```
Comment (rohitpaulk) [inline docs/challenges/language-support/csharp.mdx:52]:
```suggestion
If you want to define a class under a _different_ namespace, you'll need to use the [`namespace`](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/namespace) keyword like this:
```

(More important to highlight the `namespace` keyword than the braces)

File: docs/challenges/language-support/csharp.mdx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    }
+}
+```
+
+```csharp src/Program.cs
+using AnotherNamespace
+
+public class Main {
+    public static void main(String[] args) {
+      Foo.foo();  // Uses the foo function from Foo class defined under AnotherNamespace
```
Comment (rohitpaulk) [inline docs/challenges/language-support/csharp.mdx:83]:
```suggestion
        Foo.foo();  // Uses the foo function from Foo class defined under AnotherNamespace
```

Inconsistent indentation

File: docs/challenges/language-support/csharp.mdx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    }
+}
+```
+
+</CodeGroup>
+
+<Note>
```
Comment (rohitpaulk) [inline docs/challenges/language-support/csharp.mdx:90]: This can be removed - seems like a more nuanced detail that we don't need to cover.

**Review by rohitpaulk — APPROVED** [review]
Added minor comments, looks good otherwise!

File: docs/challenges/language-support/csharp.mdx (hunk 2 above)
Comment (glokta1) [reply docs/challenges/language-support/csharp.mdx]: CleanShot 2024-04-23 at 18 35 40@2x
@rohitpaulk Hyperlinking a code block makes the styling look wonky. Better to remove the code blocks around namespace?

**Pull request merged**